functions/prepared-check-sync/v1.js

```diff
@@ -5,18 +5,19 @@ const uuid = require('uuid/v4')
 const azureStorageHelper = require('../lib/azure-storage-helper')
 const azureTableService = azureStorageHelper.getPromisifiedAzureTableService()
 const accessArrangementsSqlUtil = require('./access-arrangements-sql-util')
-const sqlUtils = require('../lib/sql-helper')
+const checkDataService = require('./check.data.service')
 const preparedCheckSchemaValidator = require('../lib/prepared-check-schema-validator')
 
 const v1 = {}
 
 v1.process = async function process (context, preparedCheckSyncMessage) {
   const { checkCode } = preparedCheckSyncMessage
   context.log('prepared-check-sync: message received', checkCode)
-  const checks = await sqlUtils.sqlFindChecksByCheckCode(checkCode)
-  checks.map(async (check) => {
-    await v1.updatePreparedChecks(context, check.checkCode)
-  })
+  const checkCodes = await checkDataService.sqlFindActiveCheckCodesByCheckCode(checkCode)
+  for (let idx in checkCodes) {
+    await v1.updatePreparedChecks(context, checkCodes[idx])
```

Comment:
This should have a try/catch around it so that a single exception raised does not prevent the entire batch fro bein discarded.

Reply:
added